support.cpp

```diff
@@ -37,47 +37,48 @@ stripper_game_t stripper_game;
 static const char*
 parse_map(const char* map, const char* entities)
 {
-    FILE *fp;
-    char path[256];
-
-    g_Stripper.SetEntityList(entities);
-
-    stripper_game.path_format(path,
-            sizeof(path),
-            "%s/%s/global_filters.cfg",
-            stripper_game.game_path,
-            stripper_game.stripper_cfg_path);
-    fp = fopen(path, "rt");
-    if (fp == NULL)
-    {
-        stripper_game.log_message("Could not find global filter file: %s", path);
-    }
-    else
-    {
-        fclose(fp);
-        g_Stripper.ApplyFileFilter(path);
-    }
-
-    stripper_game.path_format(path,
-            sizeof(path),
-            "%s/%s/maps/%s.cfg",
-            stripper_game.game_path,
-            stripper_game.stripper_cfg_path,
-            map);
-    fp = fopen(path, "rt");
-    if (fp)
-    {
-        fclose(fp);
-        g_Stripper.ApplyFileFilter(path);
-    }
-
-    return g_Stripper.ToString();
+	FILE* fp;
+	char path[256];
+
+	g_Stripper.SetEntityList(entities);
+
+	stripper_game.path_format(path,
+		sizeof(path),
+		"%s/%s/global_filters.cfg",
+		stripper_game.game_path,
+		stripper_game.stripper_cfg_path);
+	fp = fopen(path, "rt");
+	if (fp == NULL)
+	{
+		stripper_game.log_message("Could not find global filter file: %s", path);
+	}
+	else
+	{
+		fclose(fp);
+		g_Stripper.ApplyFileFilter(path);
+	}
+
+	for (const char* pPart = map; *pPart++ != '\0';)
+	{
+		pPart = strchrnul(pPart, '_');
+		stripper_game.path_format(path, sizeof(path), "%s/%s/maps/%.*s.cfg",
+			stripper_game.game_path, stripper_game.stripper_cfg_path, pPart - map, map);
+		fp = fopen(path, "rt");
```

Comment:
It looks like this drops support for the old style of path?

Reply:
Sorry, what do you mean? The old call also used path_format. Then, the new trailing `*.*s` part is obviously directly related to the purpose of the feature.

Reply:
The old code would load map.cfg and the new code will load map.*s.cfg. Won't this prevent loading legacy paths?

Reply:
Oh, I get what you mean now. No, the `%.*s` part is a variable width specifier for strings, whose width is to be specified as an argument (`pPart - map`) and before the string pointer itself. This could possibly be a `printf` feature that wasn't years ago, I don't know, but you can search it. Anyway, in case of pattern questions that my new format line generates you can just run my REPL sample I posted originally.